recipes/flann/build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# cannot build flann from within the source directory
+mkdir build
+cd build
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+    export CC=gcc
+    export LD=gcc
+    export CXX=g++
+fi
+
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX -DBUILD_MATLAB_BINDINGS:BOOL=OFF -DBUILD_PYTHON_BINDINGS:BOOL=OFF -DBUILD_EXAMPLES:BOOL=OFF -DBUILD_DOC:BOOL=OFF
+
```

Comment:
I was meaning some general testing rule using `make`. Normally, `make check`. Though sometimes it is `make test`. After looking through this it appears `make test` is the way to go. Also, would be nice to have a call to `make` in here to. So, it would be good if we could have something like this.

    cmake <blah>
    make
    make test
    make install